src/android/src/main/java/io/radar/cordova/RadarForegroundService.java

```diff
@@ -9,18 +9,23 @@
 import android.app.NotificationManager;
 import android.os.IBinder;
 import android.os.Bundle;
+import android.util.Log;
 import android.annotation.TargetApi;
 
 public class RadarForegroundService extends Service {
 
     @Override
     public int onStartCommand(Intent intent, int flags, int startId) {
         if (intent != null) {
-            if (intent.getAction().equals("start")) {
-                startPluginForegroundService(intent.getExtras());
-            } else if (intent.getAction().equals("stop")) {
-                stopForeground(true);
-                stopSelf();
+            try {
+                if (intent.getAction().equals("start")) {
+                    startPluginForegroundService(intent.getExtras());
+                } else if (intent.getAction().equals("stop")) {
+                    stopForeground(true);
+                    stopSelf();
+                }
+            } catch (Exception e) {

```

Comment:
I've started thinking we should be doing alot more of this across the native SDKs - especially since we're going to start sending logs up to the server.